common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -506,6 +512,213 @@ def test_has_changes(self, default_ms):
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_seq_with_unpublished_vertical_has_changes_before_export_and_after_import(self, default_ms):
+        """
+        Tests that an published unit with an unpublished draft remains published across export and re-import.
+        """
+        with MongoContentstoreBuilder().build() as contentstore:
+            # initialize the mixed modulestore
+            self._initialize_mixed(contentstore=contentstore, mappings={})
+
+            with self.store.default_store(default_ms):
+
+                source_course_key = self.store.make_course_key("org.source", "course.source", "run.source")
+                self._create_course(source_course_key)
+
+                # create chapter
+                chapter = self.store.create_child(self.user_id, self.course.location, 'chapter', block_id='section_one')
+                self.store.publish(chapter.location, self.user_id)
+
+                # create sequential
+                sequential = self.store.create_child(self.user_id, chapter.location, 'sequential', block_id='subsection_one')
+                self.store.publish(sequential.location, self.user_id)
+
+                # create vertical - don't publish it!
+                vertical = self.store.create_child(self.user_id, sequential.location, 'vertical', block_id='moon_unit')
+
+                # Retrieve the published block and make sure it's published.
+                # Chapter is published - but the changes in vertical below means it "has_changes".
+                self.assertTrue(self._has_changes(chapter.location))
+                # Sequential is published - but the changes in vertical below means it "has_changes".
+                self.assertTrue(self._has_changes(sequential.location))
+                # Vertical is unpublished - so it "has_changes".
+                self.assertTrue(self._has_changes(vertical.location))
+
+                export_course_to_xml(
+                    self.store,
+                    contentstore,
+                    source_course_key,
+                    self.export_dir,
+                    'exported_source_course',
+                )
+
+                import_course_from_xml(
+                    self.store,
+                    'test_user',
+                    self.export_dir,
+                    source_dirs=['exported_source_course'],
+                    static_content_store=contentstore,
+                    target_id=source_course_key,
+                    create_if_not_present=True,
+                    raise_on_failure=True,
+                )
+
+                # Get the published xblock from the imported course.
+                # Verify that the published block still has a draft block, i.e. has changes.
+                self.assertTrue(self._has_changes(chapter.location))
+                self.assertTrue(self._has_changes(sequential.location))
+                self.assertTrue(self._has_changes(vertical.location))
```

Comment:
Optional: Would it be worth extracting the "set up data, export, import, compare" pattern as a method or a test class?